### PR TITLE
feat(perf): Add tags page prototype backend

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
This adds a prototype backend for the upcoming tags page, extending the existing endpoint with params useful for the tag page, and adding a new endpoint for performance facet histograms for a tag key.

#### Other
- This modifies histogram_query to accept an optional group_by, in addition to the filtered grouping. I'm up for counter-suggestions.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events_facets_performance.py [hunk 1]
```diff
@@ -30,6 +35,19 @@ def get(self, request, organization):
         filter_query = request.GET.get("query")
         aggregate_column = request.GET.get("aggregateColumn")
 
+        all_tag_keys = None
+        tag_key = None
+        histograms = None
+
+        if self.has_tag_page_feature(organization, request):
+            # TODO(k-fish): Temporarily allow these options for this feature, potentially circle back to split histogram off before release.
```
Comment (wmak) [inline src/sentry/api/endpoints/organization_events_facets_performance.py:43]: by this do you mean split histograms to its own endpoint?

File: src/sentry/snuba/discover.py [hunk 2]
```diff
@@ -848,16 +858,32 @@ def histogram_query(
...
+        results = query(
+            selected_columns=columns + [histogram_column, "count()"],
+            conditions=conditions,
+            query=user_query,
+            params=params,
+            orderby=[histogram_alias],
+            limit=len(fields) * num_buckets,
+            referrer=referrer,
+            functions_acl=["array_join", "histogram"],
+        )
```
Comment (wmak) [inline src/sentry/snuba/discover.py:886]: Instead could we refactor this to use `prepare_discover_query` instead? something like
```python
histogram_query = prepare_discover_query(...)
if group_by:
    histogram_query.groupby += group_by
 ```

File: src/sentry/api/endpoints/organization_events_facets_performance.py (hunk 1 above)
Comment (k-fish) [reply src/sentry/api/endpoints/organization_events_facets_performance.py:43]: Not sure, thoughts? I personally prefer separate endpoints for different data structures. What did you have in mind?

Comment (wmak) [reply src/sentry/api/endpoints/organization_events_facets_performance.py:43]: yeah IMO we shouldn't change the response structure of this endpoint based on params. so either a param for the existing histogram endpoint or a completely separate one

Comment (dashed) [reply src/sentry/api/endpoints/organization_events_facets_performance.py:43]: A separate endpoint would be amazing.

**Review by wmak — APPROVED** [review]
IMO we should put both endpoints in the same file, I'd even have them inherit from a shared base class since they share some common features like project length validation, and `query_tag_data`

**Pull request merged**